packages/valory/skills/price_estimation_abci/behaviours.py

```diff
@@ -388,6 +410,79 @@ def _get_safe_tx_hash(self) -> Generator[None, None, Optional[str]]:
         return payload_string
 
 
+class SignServiceDataHashBehaviour(PriceEstimationBaseBehaviour):
+    """Sign service data."""
+
+    matching_round = DataHashSignRound
+
+    def async_act(self) -> Generator:
+        """
+        Do the action.
+
+        Steps:
+        - Construct hash from context.shared_state[SHARED_STATE_SERVICE_DATA_KEY_NAME]
+          Sign hash
+        - Send the signature to be collected by round.
+        - Wait until ABCI application transitions to the next round.
+        - Go to the next behaviour (set done event).
+        """
+        service_data = self.context.shared_state[SHARED_STATE_SERVICE_DATA_KEY_NAME]
+
+        with self.context.benchmark_tool.measure(self.behaviour_id).consensus():
+            signature_string = yield from self.get_data_signature(service_data)
+            payload = SignaturePayload(self.context.agent_address, signature_string)
+            yield from self.send_a2a_transaction(payload)
+            yield from self.wait_until_round_end()
+        self.set_done()
+
+    def get_data_signature(self, data: Dict) -> Generator[None, None, str]:
+        """Get signature for the data."""
+        data = copy.deepcopy(data)
+
+        data.pop("agent_address", None)  # agent address is unique, need to remove it
+        data.pop("data_source", None)  # data_source can be unique, need to remove it
+
+        data_bytes = json.dumps(data, sort_keys=True).encode("ascii")
+
+        # store data to share with http handler
+        self.context.shared_state[SHARED_STATE_SERVICE_DATA_BYTES_KEY_NAME] = data_bytes
+
+        hash_bytes = hashlib.sha256(data_bytes).digest()
+
+        signature_hex = yield from self.get_signature(
+            hash_bytes, is_deprecated_mode=True
+        )
+        # remove the leading '0x'
+        signature_hex = signature_hex[2:]
+        self.context.logger.info(f"Data signature: {signature_hex}")
+        return signature_hex
+
+
+class DataHashSignatureStoreBehaviour(PriceEstimationBaseBehaviour):
+    """Save signatures for data hash in shared state to use in http server handler."""
+
+    matching_round = DataHashSignaturesStoreRound
+
+    def async_act(self) -> Generator:
+        """
+        Do the action.
+
+        Steps:
+        - Get signatures from synchronized_data
+          Store signatures into shared state
+        """
+        # store signatures to be used bu http server handler
+        self.context.shared_state[SHARED_STATE_SIGNATURES_KEY_NAME] = {
+            k.lower(): v
+            for k, v in self.synchronized_data.service_data_signatures.items()
+        }
+
+        payload = EmptyPayload(self.context.agent_address)
+        yield from self.send_a2a_transaction(payload)
```

Comment:
We can avoid this. Checkout this [behaviour](https://github.com/valory-xyz/agent-academy-1/blob/main/packages/elcollectooorr/skills/elcollectooorr_abci/behaviours.py#L1284) and its corresponding [round](https://github.com/valory-xyz/agent-academy-1/blob/main/packages/elcollectooorr/skills/elcollectooorr_abci/rounds.py#L811).

Reply:
done, thanks